src/openloco/windows/CompanyWindow.cpp

```diff
@@ -0,0 +1,1385 @@
+            {
+                uint32_t arg1{};
+                uint32_t arg2{};
+                string_id status = companymgr::getOwnerStatus(self->number, &arg1, &arg2);
+
+                *_common_format_args = status;
+                *(uint32_t*)&_common_format_args[1] = arg1;
+                *(uint32_t*)&_common_format_args[3] = arg2;
+
+                auto& widget = self->widgets[widx::unk_11];
+                gfx::draw_string_494BBF(
+                    *dpi,
+                    self->x + widget.left - 1,
+                    self->y + widget.top - 1,
+                    widget.right - widget.left,
+                    colour::black,
+                    string_ids::white_stringid2,
+                    &*_common_format_args);
+            }
+
+            if (self->viewports[0] != nullptr)
+            {
+                self->drawViewports(dpi);
+                widget::drawViewportCentreButton(dpi, self, (widget_index)widx::centre_on_viewport);
+            }
+        }
+
+        // 0x00432244
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameCompanyPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_status:
+                case common::widx::tab_details:
+                case common::widx::tab_colour_scheme:
+                case common::widx::tab_finances:
+                case common::widx::tab_cargo_delivered:
+                case common::widx::tab_challenge:
+                    common::switchTab(self, widgetIndex);
+                    break;
+
+                case widx::centre_on_viewport:
+                {
+                    if (self->viewports[0] == nullptr)
+                        break;
+
+                    // Centre viewport on tile/thing.
+                    // TODO(avgeffen): move/implement.
+                    registers regs;
+                    regs.esi = (int32_t)self;
+                    call(0x004324DC, regs);
+                    break;
+                }
+
+                case widx::face:
+                {
+                    // Open face selection window
+                    // TODO(avgeffen): move/implement.
+                    registers regs;
+                    regs.eax = self->number;
+                    call(0x00434F52, regs);
+                    break;
+                }
+
+                case widx::change_owner_name:
+                {
+                    auto company = companymgr::get(self->number);
+                    _common_format_args[2] = company->var_02;
+                    textinput::open_textinput(self, string_ids::title_name_owner, string_ids::prompt_enter_new_name_for_owner, company->var_02, widgetIndex, &_common_format_args[2]);
```

Comment:
Same with the last element being unneeded. Also, `var_02` is the owner name then.

Reply:
I was planing to amend the company struct, yes. Thanks :)